task_processing/plugins/kubernetes/kubernetes_pod_executor.py

```diff
@@ -391,10 +430,53 @@ def run(self, task_config: KubernetesTaskConfig) -> Optional[str]:
         return None
 
     def reconcile(self, task_config: KubernetesTaskConfig) -> None:
-        # TASKPROC(244): we probably only want reconcile() to fill in the task_config
-        # member of the relevant KubernetesTaskMetadata and do an unconditional state
-        # reconciliation on startup
-        pass
+        pod_name = task_config.pod_name
+        try:
+            pod = self.kube_client.get_pod(namespace=self.namespace, pod_name=pod_name)
+        except Exception:
+            logger.exception(f"Cannot reconcile pod {pod_name}")
+
+        if pod_name not in self.task_metadata:
+            self._initialize_existing_task(task_config)
+
+        with self.task_metadata_lock:
+            task_metadata = self.task_metadata[pod_name]
+            self.task_metadata = self.task_metadata.set(
+                pod_name,
+                task_metadata.set(
+                    task_config=task_config
+                )
+            )
+
+        with self.task_metadata_lock:
+            if not pod:
+                # Pod has gone away while restarting
+                logger.info(
+                    f"Pod {pod_name} for task {task_config.name} was no longer found. "
+                    "Marking as LOST"
+                )
+                self.task_metadata = self.task_metadata.set(
+                    pod_name,
+                    task_metadata.set(
+                        task_state=KubernetesTaskState.TASK_LOST,
+                        task_state_history=task_metadata.task_state_history.append(
+                            (KubernetesTaskState.TASK_LOST, time.time()),
+                        )
+                    )
+                )
+                self.event_queue.put(
+                    task_event(
+                        task_id=pod_name,
+                        terminal=False,
+                        timestamp=time.time(),
+                        raw=None,
+                        task_config=task_metadata.task_config,
+                        platform_type="lost"
+                    )
+                )
+            else:
+                # Treat like a modified pod
+                self.__update_modified_pod(pod=pod, event=None)
```

Comment:
just curious: how come we're not just nesting this in the above context manager?

Reply:
Ah, this was an error, there is no reason not to all work within the same context manager.